### PR TITLE
Android: Update gradle version to 4.1 and plugin to 3.0

### PR DESCRIPTION
**Pull request by gwicks**
Doing a simple version bump to the latest gradle version and plugin for building the android version

### CODE REVIEW

**Comment by hackbar** [comment]
@dolphin-emu-bot rebuild

File: Source/Android/build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ buildscript {
         jcenter()
```
Comment (hackbar) [inline Source/Android/build.gradle]: I think the google maven repo might need to be in buildscript. The new form in gradle 4.1 is just "google()". Replace the maven { url... down below with google() as well.

**Comment by mahdihijazi** [comment]
It could be a good idea to switch the deprecated _compile_ configuration into the new _implementation/api_ configurations in the same commit.

File: Source/Android/build.gradle (hunk 1 above)
Comment (Sarkie) [reply Source/Android/build.gradle]: correct: 
https://developer.android.com/studio/build/dependencies.html#google-maven

**Comment by hackbar** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**